orttraining/orttraining/training_ops/cuda/activation/activations_grad_impl.cu

```diff
@@ -31,6 +31,14 @@ struct OP_ReluGrad : public CtxReluGrad {
   }
 };
 
+template <typename T>
+struct OP_SigmoidGrad : public CtxSigmoidGrad {
+  __device__ __inline__ T operator()(const T& dy, const T& x) const {
+    const T sigmoid_x = (T)1 / ((T)1 + _Exp(-_Abs(x)));
```

Comment:
Is it safe to perform Exp on half type? 
We might need to cast this to float type before doing Exp

Reply:
Let's check Pytorch's impl for reference.